### PR TITLE
Fix XAML compiler handling of comma/or style selector.

### LINKED ISSUE #2838: XAMLIL compiler NRE
Using build 0.8.999-cibuild0003548-beta, I get a XAMLIL compiler NRE for the following XAML

```
Error	XAMLIL	Internal compiler error while emitting node Avalonia.Markup.Xaml.XamlIl.CompilerExtensions.Transformers.XamlIlOrSelectorNode:
System.NullReferenceException: Object reference not set to an instance of an object.
   at XamlIl.TypeSystem.XamlIlTypeSystemExtensions.EmitCall(IXamlIlEmitter emitter, IXamlIlMethod method, Boolean swallowResult) in D:\a\1\s\src\Markup\Avalonia.Markup.Xaml\XamlIl\xamlil.github\src\XamlIl\TypeSystem\TypeSystem.cs:line 474
   at Avalonia.Markup.Xaml.XamlIl.CompilerExtensions.Transformers.XamlIlSelectorNode.EmitCall(XamlIlEmitContext context, IXamlIlEmitter codeGen, Func`2 method) in D:\a\1\s\src\Markup\Avalonia.Markup.Xaml\XamlIl\CompilerExtensions\Transformers\AvaloniaXamlIlSelectorTransformer.cs:line 164
   at Avalonia.Markup.Xaml.XamlIl.CompilerExtensions.Transformers.XamlIlOrSelectorNode.DoEmit(XamlIlEmitContext context, IXamlIlEmitter codeGen) in D:\a\1\s\src\Markup\Avalonia.Markup.Xaml\XamlIl\CompilerExtensions\Transformers\AvaloniaXamlIlSelectorTransformer.cs:line 334
   a... Line 9, position 6.	AvaloniaApplication7	
...\AvaloniaApplication7.Views.ViewA.xaml	9	
```

```
<UserControl xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
             xmlns:cv="clr-namespace:AvaloniaApplication7;assembly=AvaloniaApplication7"
             mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
             x:Class="AvaloniaApplication7.Views.ViewA">
  <UserControl.Styles>
    <Style Selector="NumericUpDown,ListBox,ComboBox">
      <Setter Property="Margin" Value="5"/>
    </Style>
  </UserControl.Styles>
   <TextBox Text="{Binding UrlPathSegment}"/>
</UserControl>
```

### PR DESCRIPTION
**Pull request by grokys**
## What does the pull request do?

As described in #2838, the XAML compiler didn't handle the comma/or selector syntax. Fixes that.

- `Selector.Or`'s first parameter isn't a `Selector` - remove that check as I don't think it's needed anyway
- Make sure last selector is added to `XamlIlOrSelectorNode`

## Checklist

- [x] Added unit tests (if possible)?

## Fixed issues

Fixes #2838

### CODE REVIEW

File: tests/Avalonia.Markup.Xaml.UnitTests/Xaml/StyleTests.cs [hunk 1]
```diff
@@ -278,5 +278,67 @@ public void Style_Can_Use_Not_Selector()
...
+        [Fact]
+        public void Style_Can_Use_Or_Selector_2()
+        {
+            using (UnitTestApplication.Start(TestServices.StyledWindow))
+            {
+                var xaml = @"
+<Window xmlns='https://github.com/avaloniaui'
+             xmlns:x='http://schemas.microsoft.com/winfx/2006/xaml'>
+    <Window.Styles>
+        <Style Selector='Border,Canvas,ListBox'>
```
Comment (jmacato) [inline tests/Avalonia.Markup.Xaml.UnitTests/Xaml/StyleTests.cs]: wow.. i didnt know this was a thing already! this removes so much redundant styles in many apps out there.

**Comment by kekekeks** [comment]
`Or` selector will still not work properly though, since currently it doesn't look for a common type and uses the type of the first selector instead: https://github.com/AvaloniaUI/Avalonia/blob/33d3a66c8621f60c8d6e2fb6bab03e00a253bf86/src/Markup/Avalonia.Markup.Xaml/XamlIl/CompilerExtensions/Transformers/AvaloniaXamlIlSelectorTransformer.cs#L315

It should be OK for well-formed XAML

File: tests/Avalonia.Markup.Xaml.UnitTests/Xaml/StyleTests.cs (hunk 1 above)
Comment (robloo) [reply tests/Avalonia.Markup.Xaml.UnitTests/Xaml/StyleTests.cs]: drawing

**Comment by grokys** [comment]
@kekekeks addressed finding the common base type.

**Review by kekekeks — APPROVED** [review]
Awesome

**Pull request merged**